### ISSUE TITLE
Do not force autocorrect for hashtag spelling and capitalization

### ISSUE THREAD

**Issue by PrecisionWordcraft** [mastodon/mastodon#27013]
### Pitch

A good hashtag that adheres to accessibility principles is camel-cased, with the first letter capitalized as well, to facilitate its interpretation by screen readers. For example, #SpanishHistory is easier for screen readers to interpret than #spanishhistory.

However, if you are typing the hashtags as you create or edit a Mastodon post a browser, the UI tries to force you to accept the all-lowercase version. It will try to keep you from entering further text or from moving your cursor to another place in the post unless you accept the inaccessible hashtag. 

This is all because other people have entered inaccessible hashtags when they post, and the UI tries to steer you to one of their poorly formatted entries!  I should NOT have to fight with the UI to enter a hashtag in accessible format.

The way to end this behavior: stop the UI from trying to autocorrect hashtag spelling. 

### Motivation

People who want to create accessible post text AND people who use screen readers will benefit from this change.

**Comment by YourAutisticLife** [comment]
I'm in favor of a change in this area.

I still want autocompletion because I don't know all the tags that people have been using already. I'll note a few things here:

1. There is no central authority that decides a hashtag's spelling.

2. The same sequence of character can be validly capitalized differently depending on context. My brain ain't working right. I'm sure people can come up with better examples. I know a company called Elec Tron so the hashtag for it should be `#ElecTron` however it does not follow that people who want to talk about the particle should have their capitalization changed.

Maybe the solution would be:

1. Remember the individual user's choices and suggest *that* first.

2. Don't be so keen to autocomplete. You should autocomplete only if the user clearly requires it. Right now it is very keen to autocomplete even when I don't want it to. It could be something as simple as "show the autocompletions but don't autocomplete unless the user pressed arrows to *go* into the autocompletion dialog."

**Comment by PrecisionWordcraft** [comment]
>     2. Don't be so keen to autocomplete. You should autocomplete only if the user clearly requires it. Right now it is very keen to autocomplete even when I don't want it to. It could be something as simple as "show the autocompletions but don't autocomplete unless the user pressed arrows to _go_ into the autocompletion dialog.

This solution would take care of the problem I describe.

**Comment by AenesidemusOZ** [comment]
I agree with this suggestion. CamelCase is a far better form of hashtag, not only for screen readers but us normies as well; it's always easier to read.

Forcing a user to accept the worst format is bad for everybody.

**Comment by gunchleoc** [comment]
I am also in favor of keeping the autosuggestions and dropping the autocorrection. It's very fiddly not only when correcting to camelCase but also when your language uses diacritics (random example: `schwul` and `schwül` are two completely different words).

**Comment by YourAutisticLife** [comment]
That's a problem that I could tackle, but I don't have much inclination right now, seeing as I have gotten no traction on [this pull request.](https://github.com/mastodon/mastodon/pull/26380) (No, cryptic messages from a bot won't cut it.)

**Comment by brendanjones** [comment]
Actually duplicate of https://github.com/mastodon/mastodon/issues/25654, though this issue is more popular.
